dkeyserver/Enclave/enclave.cpp

```diff
@@ -387,7 +335,368 @@ int handle_communication_until_done(
+            errno);
+        ocall_close(&ret, sockfd);
+        if (ret != 0)
+            log_d(TLS_SERVER "OCALL: error closing socket\n");
+        sockfd = -1;
+        goto done;
+    }
+    log_d(TLS_SERVER "connected to target server %s:%d\n", server_name, server_port);
+
+done:
+    return sockfd;
+}
+
+int get_dkey_from_target(SSL *ssl)
+{
+    unsigned char buf[200];
+    int ret = 1;
+    int error = 0;
+    int len = 0;
+    int bytes_written = 0;
+    int bytes_read = 0;
+    // sgx_status_t ret = SGX_ERROR_UNEXPECTED;
+
+    // Write an GET request to the target server
+    log_d(TLS_SERVER "-----> Write to server:\n");
+    len = snprintf((char *)buf, sizeof(buf) - 1, CLIENT_PAYLOAD);
+
+    while ((bytes_written = SSL_write(ssl, buf, (size_t)len)) <= 0)
+    {
+        error = SSL_get_error(ssl, bytes_written);
+        if (error == SSL_ERROR_WANT_WRITE)
+            continue;
+        log_d(TLS_SERVER "Failed! SSL_write returned %d\n", error);
+        ret = bytes_written;
+        goto done;
+    }
+
+    log_d(TLS_SERVER "%d bytes written\n", bytes_written);
+
+    // Read the HTTP response from target server
+    log_d(TLS_SERVER "<---- Read from server:\n");
+    do
+    {
+        len = sizeof(buf) - 1;
+        memset_s(buf, sizeof(buf), 0, sizeof(buf));
+        bytes_read = SSL_read(ssl, buf, (size_t)len);
+
+        if (bytes_read <= 0)
+        {
+            int error = SSL_get_error(ssl, bytes_read);
+            if (error == SSL_ERROR_WANT_READ)
+                continue;
+
+            log_d(TLS_SERVER "Failed! SSL_read returned error=%d\n", error);
+            ret = bytes_read;
+            break;
+        }
+
+        log_d(TLS_SERVER " %d bytes read\n", bytes_read);
+
+        if (bytes_read != SGX_DOMAIN_KEY_SIZE)
+        {
+            log_d(
+                TLS_SERVER  "ERROR: expected reading %lu bytes but only "
+                           "received %d bytes\n",
+                SGX_DOMAIN_KEY_SIZE,
+                bytes_read);
+            ret = bytes_read;
+            break;
+        }
+        else
+        {
+            log_d(TLS_SERVER
+                  " received all the expected data from server\n\n");
+            ret = 0;
+            uint8_t tmp_dk[SGX_DOMAIN_KEY_SIZE] ;
+            uint32_t dk_cipher_len = sgx_calc_sealed_data_size(0, SGX_DOMAIN_KEY_SIZE);
+            uint8_t dk_cipher[dk_cipher_len] = {0};
+            memcpy_s(tmp_dk, SGX_DOMAIN_KEY_SIZE, buf, SGX_DOMAIN_KEY_SIZE);
+	    memset_s(buf, SGX_DOMAIN_KEY_SIZE, 0, SGX_DOMAIN_KEY_SIZE);
+
+            ret = sgx_seal_data(0, NULL, SGX_DOMAIN_KEY_SIZE, tmp_dk, dk_cipher_len, (sgx_sealed_data_t *)dk_cipher);
+            if (ret != SGX_SUCCESS)
+                return SGX_ERROR_UNEXPECTED;
+
+            ret = ocall_store_domain_key(&ret, dk_cipher, dk_cipher_len);
+            if (ret != SGX_SUCCESS || ret != 0)
+                return SGX_ERROR_UNEXPECTED;
+
+            log_d(TLS_SERVER "domainkey received succeed:\n");
```

Comment:
don't print it or add this logic into DEBUG build

Reply:
done.